src/validation.cpp

```diff
@@ -860,7 +860,8 @@ static bool WriteBlockToDisk(const CBlock &block, FlatFilePos &pos,
     return true;
 }
 
-Amount GetBlockSubsidy(int nHeight, const Consensus::Params &consensusParams) {
+Amount GetBlockSubsidy(uint32_t nBits,
```

Comment:
What about taking the block itself? Though, it doesn't matter much I suppose. Can be changed again if we have to hardfork.

Reply:
Makes it harder to test tbh. I thought about it but I'd rather keep it so it's just what we absolutely need; can refactor later if we need more, it is only used at like 3 places anyway.

Reply:
Yeah, I see this. This is good.